src/Components/Facility/FacilityUsers.tsx

```diff
@@ -79,32 +76,19 @@ export default function FacilityUsers(props: any) {
       }
     }
     fetchFacilityName();
-  }, [dispatch, facilityId]);
-
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-      const res = await dispatch(
-        getFacilityUsers(facilityId, { offset, limit })
-      );
-
-      if (!status.aborted) {
-        if (res && res.data) {
-          setUsers(res.data.results);
-          setTotalCount(res.data.count);
-        }
-        setIsLoading(false);
+  }, [facilityId]);
+
+  const { refetch } = useQuery(routes.getFacilityUsers, {
+    query: { offset: offset, limit: limit },
+    pathParams: { facility_id: facilityId },
+    prefetch: facilityId !== undefined,
+    onResponse: ({ res, data }) => {
+      if (res?.ok && data) {
+        setUsers(data.results);
+        setTotalCount(data.count);
```

Comment:
Delete these users and total count useStates. use the data from this query and do

```jsx
const users = data?.results;
const totalCount = data?.count;
```